subprojects/docs/src/docs/release/notes.md

```diff
@@ -61,6 +61,72 @@ With this release of Gradle, the console displays any test worker processes exec
 Tasks of types [ScalaCompile](dsl/org.gradle.api.tasks.scala.ScalaCompile.html) and [ScalaDoc](dsl/org.gradle.api.tasks.scala.ScalaDoc.html) provided by the [scala](userguide/scala_plugin.html) plugin are now cacheable.
 This means they will make use of the build cache when activated.
 
+### New API for safe, fast concurrent work execution
+
+Gradle 4.1 introduces the [Worker API](userguide/custom_tasks.html#worker_api), a new API for safe, parallel, and asynchronous execution of build logic.  This API allows for:
+
+- Parallel execution of multiple items of work within a task
+- Execution of work in a separate daemon process
+- Safe intra-project parallel execution of tasks
+
+This API can be used within a custom task class to break up the work of the task and execute that work in parallel.  Once a 
+task has submitted all of its work to run asynchronously, and has exited the task action, Gradle can then start running other 
+independent tasks in parallel - even if those tasks are in the same project.  This allows Gradle to make maximum use of the 
+resources available and complete builds faster than ever.
+
+    import javax.inject.Inject
+    
+    // The implementation of a single unit of work
+    class UnitOfWork implements Runnable {
+        File fileToReverse
+        File destinationFile
+        
+        @Inject
+        public UnitOfWork(File fileToReverse, File destinationFile) {
+            this.fileToReverse = fileToReverse
+            this.destinationFile = destinationFile
+        }
+        
+        @Override
+        public void run() {
+            destinationFile.text = fileToReverse.text.reverse()
+        }
+    }
+    
+    // A task that accepts a set of source files, creates units of work for each,
+    // and executes them concurrently.
+    class ReverseFiles extends SourceTask {
+        @OutputDirectory
+        File outputDir
+        
+        // The WorkerExecutor will be injected by Gradle at runtime 
+        // (i.e. the exception below is only a placeholder and will not be thrown)
+        @Inject
```

Comment:
Yeah, it won't work without an import.  It's a little weird that something that's important to the API isn't automatically available.  I don't think it's a huge deal because it sucks to keep classes inside your build file.